packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +100,123 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
+/// children selection ranges that may be contained within the active
+/// selection, and the content itself.
+///
+/// [SelectionArea] and [SelectableRegion] provide access to this
+/// information through the [SelectedContent] provided in their
+/// [SelectionArea.onSelectionChanged] and
+/// [SelectableRegion.onSelectionChanged] members.
+///
+/// {@tool dartpad}
+/// This example shows how to color red the active selection
+/// under a [SelectionArea] or [SelectableRegion].
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.1.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to replace the active selection
+/// under a [SelectionArea] or [SelectableRegion] with a widget.
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.2.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [SelectedContent], which contains the [SelectedContentRange] for a given [Selectable] or [SelectionHandler].
+abstract class SelectedContentRange<T extends Object> {
+  /// Creates a range for the content of a [Selectable] or [SelectionHandler].
+  SelectedContentRange({
+    this.selectableId,
+    required this.content,
+    this.children,
+  });
+
+  /// The unique id for the [Selectable] that created the range.
+  final int? selectableId;
+
+  /// The content that contains the selection.
+  final T content;
+
+  /// The start of the selection relative to the [content].
+  int get startOffset;
+
+  /// The end of the selection relative to the [content].
+  int get endOffset;
+
+  /// Additional ranges to include as children.
+  ///
+  /// Children of a given range enable more granular modification of the
+  /// selection.
+  final List<SelectedContentRange<Object>>? children;
+
+  @override
+  String toString() {
+    return 'SelectedContentRange(\n'
+           '  selectableId: $selectableId,\n'
+           '  content: $content,\n'
+           '  startOffset: $startOffset,\n'
+           '  endOffset: $endOffset,\n'
+           '  children: $children,\n'
+           ')';
+  }
+}
+
 /// The selected content in a [Selectable] or [SelectionHandler].
 // TODO(chunhtai): Add more support for rich content.
 // https://github.com/flutter/flutter/issues/104206.
 class SelectedContent {
   /// Creates a selected content object.
   ///
   /// Only supports plain text.
-  const SelectedContent({required this.plainText});
+  const SelectedContent({
+    required this.plainText,
+    required this.geometry,
+    this.startOffset = -1,
+    this.endOffset = -1,
+    this.ranges,
+  });
 
   /// The selected content in plain text format.
   final String plainText;
+
+  /// The [SelectionGeometry] of the [Selectable] or [SelectionHandler] containing
+  /// the selection.
+  final SelectionGeometry geometry;
+
+  /// The value representing the beginning of the selection, defaults to -1.
+  ///
+  /// If the [Selectable] contains text, then the offset represents
```

Comment:
Someone just reading this class's docs might wonder "what Selectable?" because this class has no Selectable member.

Reply:
Good point, another reason to move these offsets somewhere else.